docs/api/sql/Raster-aggregate-function.md

```diff
@@ -13,30 +13,55 @@ Format: `RS_Union_Aggr(A: rasterColumn, B: indexColumn)`
 
 Since: `v1.5.1`
 
-SQL Example
+SQL Example:
 
-Contents of `raster_table`.
+First, define a window specification that partitions by geographic location and orders by time. This will prepare your data by assigning an index to each raster based on its timestamp within each location group.
 
 ```
-+------------------------------+-----+
-|                        raster|index|
-+------------------------------+-----+
-|GridCoverage2D["geotiff_cov...|    1|
-|GridCoverage2D["geotiff_cov...|    2|
-|GridCoverage2D["geotiff_cov...|    3|
-|GridCoverage2D["geotiff_cov...|    4|
-|GridCoverage2D["geotiff_cov...|    5|
-+------------------------------+-----+
+val windowSpec = Window.partitionBy("geometry").orderBy("timestamp")
+val indexedRasters = df.withColumn("index", row_number().over(windowSpec))
+
+indexedRasters.show()
+```
+
+The indexed rasters will appear as follows, showing that each raster is tagged with a sequential index (ordered by timestamp) within its group (geometry).
+
+```
++-------------------+------------------------------+--------------+-----+
+|          timestamp|                        raster|      geometry|index|
++-------------------+------------------------------+--------------+-----+
+|2021-01-01T00:00:00|GridCoverage2D["geotiff_cov...|POINT (72 120)|    1|
+|2021-01-02T00:00:00|GridCoverage2D["geotiff_cov...|POINT (72 120)|    2|
+|2021-01-03T00:00:00|GridCoverage2D["geotiff_cov...|POINT (72 120)|    3|
+|2021-01-04T00:00:00|GridCoverage2D["geotiff_cov...|POINT (72 120)|    4|
+|2021-01-05T00:00:00|GridCoverage2D["geotiff_cov...|POINT (72 120)|    5|
+|2021-01-02T00:00:00|GridCoverage2D["geotiff_cov...|POINT (84 132)|    1|
+|2021-01-03T00:00:00|GridCoverage2D["geotiff_cov...|POINT (84 132)|    2|
+|2021-01-04T00:00:00|GridCoverage2D["geotiff_cov...|POINT (84 132)|    3|
+|2021-01-05T00:00:00|GridCoverage2D["geotiff_cov...|POINT (84 132)|    4|
+|2021-01-06T00:00:00|GridCoverage2D["geotiff_cov...|POINT (84 132)|    5|
+|2021-01-07T00:00:00|GridCoverage2D["geotiff_cov...|POINT (84 132)|    6|
++-------------------+------------------------------+--------------+-----+
 ```
 
+To create a stacked raster by grouping on geometry.
+
 ```
-SELECT RS_Union_Aggr(raster, index) FROM raster_table
+SELECT geometry, RS_Union_Aggr(raster, index) AS raster, RS_NumBands(raster) AS Num_Bands
+FROM indexedRasters
+WHERE index <= 4
+GROUP BY geometry
 ```
 
 Output:
 
-This output raster contains the first band of each raster in the `raster_table` at specified index.
+The query yields rasters grouped by geometry, each containing the first four time steps combined into a single multiband raster, where each band represents one time step.
```

Comment:
I think this is a great example and sufficient for merging. but can it be followed up by a PR showing how to groupby different time intervals, instead of or in addition to taking the first four timesteps? 

For example, we might want to group by a single year, then take the first 4 timesteps in that specific year.


Or, we might also want to sample scenes at a monthly, or year-quarterly frequency. Would be good to show an example of group by interval that matches pandas and sample interval

pandas example:

```
import pandas as pd
df['date'] = pd.to_datetime(df['date'])
df.set_index('date', inplace=True)

# Group by year and sample every quarter
quarterly_sample = df.resample('A').asfreq('Q')

# Group by year and sample every month
monthly_sample = df.resample('A').asfreq('M')
```

Reply:
Thanks Ryan! I see your point, this is definitely a more realistic example. I've updated the example to show a groupby over a quarterly time interval in addition to geometry.